docs/decisions/0003-lti-1p3-score-linking.rst

```diff
@@ -0,0 +1,96 @@
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Option name
+     - Behavior
+   * - Only allow tools to push a single grade per student (Default)
+     - This is the "declarative" approach, where the platform will create a single LineItem, and only allow read-only access to it.
+       The tool is only able to read LineItems, push scores and retrieve grades.
+   * - Allow tools to manage their own grades
+     - This is the "programmatic" approach, where tools have full permissions to create, edit and delete LineItems, as well as
+       pushing and retrieving grades. Note that there might be cases where a tool doesn't set a `resouceLinkId` leaving the grade
+       unlinked to a problem. Also, the other edge case is when a tool pushes multiple grades for a single problem, in which case
+       the implementation needs to decide on a criteria to merge the grades or pick one.
+
+Declarative grade handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+When the LTI configuration is created, a signal should trigger the creation of a LineItem if the tool is configured to use the declarative
+model. The LineItem created will have the following attributes:
+
+.. list-table::
+   :widths: auto
+   :header-rows: 1
+
+   * - Attribute
+     - Value
+   * - lti_configuration
+     - LTI configuration just created.
+   * - resource_id
+     - Blank, this is used by LTI tools in the programmatic interaction model.
+   * - label
+     - The problem title, derived from the block's attributes.
+   * - score_maximum
+     - Maximum score for this given problem, derived from the block's attributes.
+   * - tag
+     - Blank, this is used by LTI tools in the programmatic interaction model.
+   * - start_date_time
+     - The problem's start date, if available in the block's attributes.
+   * - end_date_time
+     - The problem's end date, if available in the block's attributes.
+
+Programmatic grade handling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+No LineItems are created when the LTI configuration model is created, but the tools have permissions to create, update and
+delete LineItems using the LineItem endpoint.
+
+A *post_save* Django signal in the *LtiAgsScore* should be responsible for loading the XBlock from the modulestore,
+bind the user to the session, and set the score (after doing the proper scaling using the *scoreMaximum* attribute).
+
+If a tool creates and links multiple LineItems to the same problem, the platform will merge the results as follows:
+1. When a *Score* is pushed, the post save signal checks if the grade submitted is the final grade (by checking that *gradingProgress* is *FullyGraded*).
+2. The signal queries for other *LineItems* **linked to the problem** (with *LtiResourceLinkId* set) and checks if there's any fully graded *Score* for the user.
+3. With the just created *Score* plus the others previously submitted retrieved, the platform will normalize the values (using *scoreMaximum*) and average them.
```

Comment:
In step 1, does this mean nothing happens if the grade is not FullyGraded? That is, steps 2 and 3 only happen if the grade is FullyGraded.
In step 3, only FullyGraded values are averaged?

Reply:
We should also note here that this is a choice we can revisit later.  In the fullness of time, based on demand, we could (hypothetically) provide the instructor with a choice of how to handle multiple grades.